Update IPOPT and HSL documentation for Kestrel (#697)
NREL/HPC: docs/Documentation/Applications/ipopt.md

IPOPT is commonly used in solving power flow, e.g., AC Optimal Power Flow, and controls problems. Please refer to their [project website](https://github.com/coin-or/Ipopt) for the source code. The documentation can be found [here](https://coin-or.github.io/Ipopt/index.html).

!!! note

>     IPOPT with HSL linear solvers is available as a module on Kestrel. Please see [IDAES Solvers](./idaes_solvers.md) for additional details. We recommend using the system module for ease-of-use and only build if they do not meet your needs.

Reviewer: ```suggestion
    IPOPT with HSL linear solvers is available as a module on Kestrel. Please see [IDAES Solvers](./idaes_solvers.md) for additional details. We recommend using the system module for ease-of-use and only build if the module does not meet your needs.
```

## Installation from source

!!! info

>     We advise build all applications on the compute node on an interactive session. Please see [Running Interactive Jobs](../Slurm/interactive_jobs.md#running-interactive-jobs) for additional details.

Reviewer: ```suggestion
    We advise building all applications on a compute node using an interactive session. Please see [Running Interactive Jobs](../Slurm/interactive_jobs.md#running-interactive-jobs) for additional details.
```

### Optional Pre-requisites

We will build IPOPT using all prerequisites mentioned below. Users may pick and
choose depending on their needs.

#### Metis

> It is highly recommended installing [Metis](https://github.com/KarypisLab/METIS.git)

Reviewer: ```suggestion
It is highly recommended to install [Metis](https://github.com/KarypisLab/METIS.git)
```

- Serial Graph Partitioning and Fill-reducing Matrix Ordering software to 
improve the performance of linear solvers such as MUMPS and HSL.

!!! warning
    Using HSL linear solvers requires installing Metis. Metis is optional for MUMPS.

We will install Metis using Anaconda. However, it can also be installed from source.
To install using Anaconda, we will create a clean environment with only Metis.

> The conda environment is being constructed within a directory in the `hpcapps` project on 

Reviewer: ```suggestion
For this example, the conda environment is being constructed within a directory in the `hpcapps` project on 
```
